components/event-signup.js

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import { useState, useEffect } from "react";
+
+function EventSignup(eventProps) {
+  const [eventData, setEventData] = useState();
+  const [registerSuccess, setRegisterSuccess] = useState();
+  const [ticketUrl, setTicketUrl] = useState("");
+
+  useEffect(() => {
+    checkSlug();
+  }, []);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful, isSubmitting },
+  } = useForm({
+    mode: "onTouched",
+  });
+
+  const checkSlug = async () => {
+    const eventData = fetch("/api/tito-check-slug")
+      .then((response) => response.json())
+      .then((json) => {
+        setEventData(json);
+      })
+      .catch((error) => {
+        res.json(error);
+        res.status(405).end();
```

Comment:
`res` will be undefined. You might be looking for a error state to set.

Reply:
Resolved in commit: 5d5cec91450b80a76cfbc9667e45672d71462065

This function doesn't exist in this component anymore so the error handling is gone with it.